pkgs/top-level/all-packages.nix

```diff
@@ -17977,8 +17978,9 @@ with pkgs;
   electron_28 = if lib.meta.availableOn stdenv.hostPlatform electron-source.electron_28 then electron-source.electron_28 else electron_28-bin;
   electron_29 = if lib.meta.availableOn stdenv.hostPlatform electron-source.electron_29 then electron-source.electron_29 else electron_29-bin;
   electron_30 = if lib.meta.availableOn stdenv.hostPlatform electron-source.electron_30 then electron-source.electron_30 else electron_30-bin;
-  electron = electron_29;
-  electron-bin = electron_29-bin;
+  electron_31 = if lib.meta.availableOn stdenv.hostPlatform electron-source.electron_31 then electron-source.electron_31 else electron_31-bin;
+  electron = electron_31;
+  electron-bin = electron_31-bin;
```

Comment:
Not a fan of this when backporting to stable. This will probably cause some regressions.

Reply:
I'm doing this because 29 [is EOL today](https://www.electronjs.org/docs/latest/tutorial/electron-timelines). I can revert the change if using an EOL version of electron as the default is fine on stable.

Reply:
Would be nice to reach some consensus on this. We have this disucssion every time an Electron major release reaches end-of-life :sweat_smile:

Reply:
I don’t think it’s okay from a security perspective to keep packages widely depending on EOL browser engines. Those are already an established exception to stability.

Reply:
Then we probably need to bump a handful of packages depending on electron with breaking changes.

Reply:
The alternative is `knownVulnerabilities`.

Reply:
Setting knownVulnerabilities on electro/chromium built from source is effectively throwing people a stick into their bike wheels and showing them a middle finger. IMO we should just built packages with meta.knownVulnerabilities.

Reply:
We should not waste limited Hydra cycles building a long tail of broken and insecure software; we should be removing more of it instead. We have a special list of exemptions for insecure software that Hydra can build if necessary, but the correct thing to do is to move packages off this EOL version so that users are safe, as is happening in various PRs now.

Anyway this policy discussion is completely off‐topic for this pull request.